### PR TITLE
fix: hubspot signup form issue when user is invited

### PR DESCRIPTION
**Pull request by jainpawan21**
### What change does this PR introduce?
1. Add new API to update user profile data (firstName, lastName and jobTitle)
2. Fix the bug of hubspot form not showing up when new user is invited to join the org

### CODE REVIEW

File: apps/web/src/pages/auth/InvitationPage.tsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ export default function InvitationPage() {
   useEffect(() => {
     // auto accept invitation when logged in as invited user
     if (isLoggedInAsInvitedUser) {
-      submitToken(tokensRef.current.token as string, tokensRef.current.invitationToken as string, true);
+      submitToken(tokensRef.current.token as string, tokensRef.current.invitationToken as string, true, false);
```
Comment (LetItRock) [inline apps/web/src/pages/auth/InvitationPage.tsx:48]: it's hard to understand what are there `true/false` arguments, please change it to the object, which will improve readability

File: apps/web/src/pages/auth/components/HubspotSignupForm.tsx [hunk 2]
```diff
@@ -17,39 +17,52 @@ import SetupLoader from './SetupLoader';
...
 
           return;
         }
-
-        navigate(ROUTES.HOME);
       }
     }
-  }, [token, navigate, isFromVercel, startVercelSetup]);
+
+    return () => {};
```
Comment (LetItRock) [inline apps/web/src/pages/auth/components/HubspotSignupForm.tsx:60]: not needed

File: apps/web/src/pages/auth/components/useAcceptInvite.ts [hunk 3]
```diff
@@ -12,15 +12,15 @@ import { errorMessage } from '../../../utils/notifications';
...
   const queryClient = useQueryClient();
+  const navigate = useNavigate();
 
   const { isLoading, mutateAsync, error, isError } = useMutation<string, IResponseError, string>((tokenItem) =>
     api.post(`/v1/invites/${tokenItem}/accept`, {})
   );
 
   const submitToken = useCallback(
-    async (token: string, invitationToken: string, refetch = false) => {
+    async (token: string, invitationToken: string, refetch = false, isSignUp = true) => {
```
Comment (LetItRock) [inline apps/web/src/pages/auth/components/useAcceptInvite.ts:23]: by default `isSignUp` should be `false`, we only have it set to `true` in one place - sign up form

File: apps/web/src/pages/auth/components/SignUpForm.tsx [hunk 4]
```diff
@@ -77,9 +77,7 @@ export function SignUpForm({ invitationToken, email }: SignUpFormProps) {
     applyToken(token);
 
     if (invitationToken) {
-      submitToken(token, invitationToken);
-
-      return true;
```
Comment (LetItRock) [inline apps/web/src/pages/auth/components/SignUpForm.tsx:82]: why did we remove this `return` statement?

**Review by LetItRock — CHANGES_REQUESTED** [review]
hey @jainpawan21 👋 

Please make sure to verify these cases:
1. Invited the existing user (registered by email) that is currently logged, the form should not be shown.
2. Invited the existing user (registered by email) that is not logged in, the form should not be shown.
3. Invited the existing user (registered by GH) that is currently logged, the form should not be shown.
4. Invited the existing user (registered by GH) that is not logged in, the form should not be shown.
5. Invited the new user register by email, the form should be shown.
6. Invited the new user register by GH, the form should be shown.

The case number four is failing, attaching the video.

https://github.com/novuhq/novu/assets/2607232/6b06cc7e-5afa-4d1d-85b2-eea64b8b421e

I would appreciate if you can cover the new logic with Cypress E2E tests, as far as I see we are lucking tests for the invitation logic, but it would be much appreciated if we can add them. Excluding GH flow as it's complex and requires account confirmation in E2E tests.

**Comment by scopsy** [comment]
@jainpawan21 can we push to merge or close this PR? 🙏

**Comment by jainpawan21** [comment]
> @jainpawan21 can we push to merge or close this PR? 🙏

Closing this PR. I will create a new PR with requested changes.

**Pull request closed without merging**